plugins/inputs/fritzbox/fritzbox.go

```diff
@@ -0,0 +1,471 @@
+	fields["errored_secs"] = statisticsTotal.NewErroredSecs
+	fields["severly_errored_secs"] = statisticsTotal.NewSeverelyErroredSecs
+	fields["fec_errors"] = statisticsTotal.NewFECErrors
+	fields["atuc_fec_errors"] = statisticsTotal.NewATUCFECErrors
+	fields["hec_errors"] = statisticsTotal.NewHECErrors
+	fields["atuc_hec_errors"] = statisticsTotal.NewATUCHECErrors
+	fields["crc_errors"] = statisticsTotal.NewCRCErrors
+	fields["atuc_crc_errors"] = statisticsTotal.NewATUCCRCErrors
+	acc.AddFields("fritzbox_dsl", fields, tags)
+	if plugin.Debug {
+		plugin.logMetric("fritzbox_dsl", tags, fields)
+	}
+	return nil
+}
+
+func (plugin *Fritzbox) gatherWlanInfo(acc telegraf.Accumulator, deviceClient *tr064.Client, service tr064.ServiceDescriptor) error {
+	serviceClient := wlanconfig.ServiceClient{
+		TR064Client: deviceClient,
+		Service:     service,
+	}
+	info := &wlanconfig.GetInfoResponse{}
+	err := serviceClient.GetInfo(info)
+	if err != nil {
+		return err
+	}
+	if info.NewStatus != "Up" {
+		return nil
+	}
+	totalAssociations := &wlanconfig.GetTotalAssociationsResponse{}
+	err = serviceClient.GetTotalAssociations(totalAssociations)
+	if err != nil {
+		return err
+	}
+	tags := make(map[string]string)
+	tags["fritz_device"] = serviceClient.TR064Client.DeviceUrl.Hostname()
+	tags["fritz_service"] = serviceClient.Service.ShortId()
+	tags["fritz_wlan"] = info.NewSSID
+	tags["fritz_wlan_channel"] = strconv.Itoa(int(info.NewChannel))
+	tags["fritz_wlan_band"] = plugin.wlanBandFromInfo(info)
+	fields := make(map[string]interface{})
+	fields["total_associations"] = totalAssociations.NewTotalAssociations
+	acc.AddGauge("fritzbox_wlan", fields, tags)
+	if plugin.Debug {
+		plugin.logMetric("fritzbox_wlan", tags, fields)
+	}
+	return nil
+}
+
+func (plugin *Fritzbox) wlanBandFromInfo(info *wlanconfig.GetInfoResponse) string {
+	band := info.NewX_AVM_DE_FrequencyBand
+	if band != "" {
+		return band
+	}
+	if 1 <= info.NewChannel && info.NewChannel <= 14 {
+		return "2400"
+	}
+	return "5000"
+}
+
+func (plugin *Fritzbox) gatherHostsInfo(acc telegraf.Accumulator, deviceClient *tr064.Client, service tr064.ServiceDescriptor) error {
+	if plugin.queryCycleCounter != 0 {
+		return nil
+	}
+	serviceClient := hosts.ServiceClient{
+		TR064Client: deviceClient,
+		Service:     service,
+	}
+	connections, err := plugin.fetchHostsConnections(&serviceClient)
+	if err != nil {
+		return err
+	}
+	for _, connection := range connections {
+		_, err = uuid.Parse(connection.RightDeviceName)
+		if err == nil {
+			continue
+		}
+		tags := make(map[string]string)
+		tags["fritz_device"] = serviceClient.TR064Client.DeviceUrl.Hostname()
+		tags["fritz_service"] = serviceClient.Service.ShortId()
+		tags["fritz_host"] = connection.RightDeviceName
+		tags["fritz_host_role"] = plugin.hostRole(connection.RightMeshRole)
+		tags["fritz_host_peer"] = connection.LeftDeviceName
+		tags["fritz_host_peer_role"] = plugin.hostRole(connection.LeftMeshRole)
```

Comment:
What about keeping the left and right naming?
```suggestion
		tags["right_device"] = connection.RightDeviceName
		tags["right_role"] = plugin.hostRole(connection.RightMeshRole)
		tags["left_device"] = connection.LeftDeviceName
		tags["left_role"] = plugin.hostRole(connection.LeftMeshRole)
```

Reply:
The left and right naming is coming from the generic connection struct. In current context only specific connections between host and the corresponding (mesh) access points are listed. I will adjust the names to host/host_role and host_ap/host_ap_role to better reflect these semantics.

Reply:
Done via 956766ca069329ce108d2513ab3f4d970f249921